### ISSUE TITLE
Map out certbot-auto burial

### ISSUE THREAD

**Issue by bmw** [certbot/certbot#8526]
With https://github.com/certbot/certbot/pull/8489, certbot-auto is fully deprecated, but I think there's a little more work to be done. I think the biggest thing here is the script is still checking for updates on every run and we have to provide a signed version of the script with every release to prevent it from breaking. I think we should change this, although exactly when and how we do that are unresolved questions in mind.

The other work to be done here is largely looking through every reference to `certbot-auto` or `letsencrypt-auto` in this repo and deciding what to do with it. While we're here, let's kill this thing properly and not leave behind a bunch of cruft.

This issue is just tracking out making plans and mapping out the work here. This is usually something I do myself, but I'm pretty busy and there's no reason others can't help with this. I think most of this work should not actually land in `master` before we ship https://github.com/certbot/certbot/pull/8489, however, planning this work or even getting PRs ready could be done now.

I think this is important so I'm putting this issue in the milestone for the certbot-auto deprecation so we can start thinking about this for our upcoming releases.

**Comment by bmw** [comment]
I was considering writing up a longer and slightly more formal proposal to modifying the certbot-auto update mechanism, but I figured I'd share what I've been thinking about now in case any one else is able to help here.

My current idea is to make use of the long supported `NO_SELF_UPGRADE` environment variable. I think if we set `NO_SELF_UPGRADE=1` somewhere in the script, once users update to a version of the script that does that, they'll stop checking for updates. In some ways, this is a scary point of no return and I think we probably should do this by OS over 2-3 releases to minimize potential problems, however, I don't really expect us to hit many issues here. `NO_SELF_UPGRADE` as set by the `--no-self-upgrade` flag has been supported and used for a very long time.

With this plan, once we're setting `NO_SELF_UPGRADE=1` for every OS, we can just leave letsencrypt-auto and it's signature as fetched by the script unmodified in our repo and remove any code making changes to it during a release. We'd need to keep around these files forever or at least the foreseeable future and we should probably have some tests to help ensure that, but I personally think this is fine and this overall plan is the one I personally like the most so far.
